misk-crypto/src/main/kotlin/misk/crypto/KeyManager.kt

```diff
@@ -67,4 +69,56 @@ class MacKeyManager @Inject internal constructor(
     }
     return mac
   }
-}
\ No newline at end of file
+}
+
+/**
+ * Holds a map of every key name to its corresponding [PublicKeySign] and [PublicKeyVerify] primitives.
+ * Users may use this class to obtain a [PublicKeySign] to sign data
+ * or [PublicKeyVerify] to verify the integrity of a message dynamically:
+ * ```
+ * val signer: PublicKeySign? = keyManager.getSigner("myDigitalSignatureKey")
+ * val verifier: PublicKeyVerify? = keyManager.getVerifier("mySigitalSignatureKey")
+ * val signature = signer.sign(data)
+ * verifier.verify(signature, data)
+ * ```
+ */
+@Singleton
+class DigitalSignatureKeyManager @Inject internal constructor(
+  private val injector: Injector
+) {
+  private val signers: HashMap<String, DigitalSignature> = LinkedHashMap()
+
+  internal operator fun set(name: String, signerAndVerifier: DigitalSignature) {
+    if (!signers.containsKey(name)) {
+      signers[name] = signerAndVerifier
+    }
+  }
+
+  fun getSigner(name: String): PublicKeySign? {
+    var signer = signers[name]?.signer
+    if (signer == null) {
+      try {
+        signer = injector.getInstance(Key.get(PublicKeySign::class.java, Names.named(name)))
+        val verifier = injector.getInstance(Key.get(PublicKeyVerify::class.java, Names.named(name)))
+        this[name] = DigitalSignature(signer, verifier)
+      } catch (e: ConfigurationException) {
```

Comment:
should we be logging or exploding more loudly here? When would this happen?

Reply:
It shouldn't, but any call to `Injector:getInstance` might produce a `ConfigurationException` and should be wrapped with a `try/catch`.

I know Kotlin has some nicer ways/tricks to deal with checked exceptions - is there anything you think I should change here?

Reply:
hmm, I don't think `ConfigurationException` is a checked exception, can you try removing this?